### ISSUE TITLE
[BUG] TypeError: domain.enter is not a function/ Browser.close() Promise does not resolve when "domain" package is used

### ISSUE THREAD

**Issue by mxschmitt** [microsoft/playwright#3848]
**Context:**
- Playwright Version: 1.4
- Operating System: Mac
- Node.js version: 12 / 13
- Browser: All
- Extra: Originally from Try Playwright

**Code Snippet**

```js
const domain = require("domain")
const playwright = require("playwright");

(async () => {
  const local = domain.create();
  local.run(() => { })
  const browser = await playwright.chromium.launch();
  const context = await browser.newContext();
  await context.newPage();
  console.log("before-block")
  await browser.close();
  console.log("after-block")
})();
```

With express Sentry plugin:

```javascript
const express = require('express')
const Sentry = require('@sentry/node')
const playwright = require("playwright");

const app = express()

// after removing it will work again
app.use(Sentry.Handlers.requestHandler())

app.get("/", async (req, resp) => {
  try {
    const browser = await playwright.chromium.launch();
    const context = await browser.newContext();
    await context.newPage();
    console.log("before-block")
    await browser.close();
    console.log("after-block")
  } catch (error) {
    console.log(error)
  }
})

app.listen(3001, () => {
  console.log("curl -v http://localhost:3001")
})
```

Their "domain" implementation: https://github.com/getsentry/sentry-javascript/blob/7c142839036e77161a6f707196762eb718b723cc/packages/node/src/handlers.ts#L272-L281

On Heroku:

![image](https://user-images.githubusercontent.com/17984549/92896101-5d4fe480-f41c-11ea-803e-3348cef3ba3b.png)

**Describe the bug**

Browser.close() blocks. The "after block" message won't get logged. This has probably something to do with this [Slack message](https://playwright.slack.com/archives/C0141EZ6ZNH/p1599696759002200).

Maybe upstream? https://github.com/nodejs/node/issues/34720
Relates https://github.com/getsentry/sentry-javascript/issues/1843

But not sure why it did not happen in 1.3.

**Comment by vrknetha** [comment]
I am getting the same error once i upgrade to 1.4. I do not use the domain package anywhere but still getting the error

Playwright: 1.4
OS: Windows and Ubuntu 18.04

![image](https://user-images.githubusercontent.com/20596261/92896878-60f26400-f43a-11ea-84f0-ca64e568b93b.png)

**Comment by sergeyt** [comment]
getting this error if using `ts-node` to run a test script

**Comment by mxschmitt** [comment]
> getting this error if using `ts-node` to run a test script

Yep we didn't release a version yet with this fix. As a workaround you can use the bleeding edge version: `npm install playwright@next` until we released 1.4.1.

**Comment by sergeyt** [comment]
Another workaround. Compile TS into JS and run compiled code.
